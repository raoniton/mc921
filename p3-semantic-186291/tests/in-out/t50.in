// Derived class
class Dog extends Animal {
    String breed;
    String name;

    // Override method
    public void makeSound() {
        print("Woof!");
    }

    // New method
    public void fetch() {
        print(this.name, " is fetching.");
    }
}


// Main class
class Main {
    public static void main(String[] args) {
        // Create an instance of Dog
        Dog myDog = new Dog();
        myDog.age = 3;
        myDog.name = "Buddy";
        myDog.breed = "Golden Retriever";
        myDog.makeSound(); // Output: Woof!
        myDog.fetch();     // Output: Buddy is fetching.
    }
}