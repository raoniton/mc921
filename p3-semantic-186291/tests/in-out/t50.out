SemanticError: Animal is not a class type @ 2:1
